lib/src/code_field/code_controller.dart

```diff
@@ -380,6 +384,121 @@ class CodeController extends TextEditingController {
     });
   }
 
+  /// Comments or uncomments the current selected lines.
+  ///
+  /// Doesn't affect empty lines
+  ///
+  /// If any of the selected lines is not a single line comment:
+  /// adds one level of single line comment to every selected line.
+  ///
+  /// If all of the selected lines are single line comments:
+  /// removes one level of single line comment from every selected line.
+  ///
+  /// When commenting adds `// ` or `# ` (or another symbol depending on a language) with a space after.
+  /// Removes these spaces on uncomment respectively.
+  ///
+  /// The method doesn't care about multiline comments
+  /// and treats them as a normal text (not a comment)
+  void commentOrUncommentSelection() {
+    final commentTypes = SingleLineComments.byMode[language] ?? [];
+
+    for (final commentType in commentTypes) {
+      final regExpForCommentedLine =
+          RegExps.getSingleLineCommentRegExp(commentType, multiline: true);
+
+      // line is either a comment or empty
+      final allSelectedLinesAreCommented = checkSelectedLinesForCondition(
+        (line) =>
+            regExpForCommentedLine.hasMatch(line) ||
+            RegExps.emptyLine.hasMatch(line),
+      );
+
+      if (allSelectedLinesAreCommented) {
+        _uncommentSelectedLines(commentType);
+        return;
+      } else {
+        _commentSelectedLines(commentType);
+        return;
+      }
+    }
+  }
+
+  /// Utility method to divide [commentOrUncommentSelection] method into parts
+  /// [commentSymbols] - `//` or `#` indicates what to use to comment
+  void _commentSelectedLines(String commentSymbols) {
+    int? startOfCommentSymbols;
+
+    modifySelectedLines((line) {
+      // if line is empty do not comment it
+      if (RegExps.emptyLine.hasMatch(line)) {
+        return line;
+      }
+
+      startOfCommentSymbols ??=
+          RegExps.whiteSpacesAfterStartOfLine.firstMatch(line)?.end;
+
+      if (startOfCommentSymbols == null) {
+        return '$commentSymbols $line';
+      }
+
+      return line.replaceRange(
+        startOfCommentSymbols!,
+        startOfCommentSymbols,
+        '$commentSymbols ',
+      );
+    });
+  }
+
+  /// Utility method to divide [commentOrUncommentSelection] method into parts
+  /// [commentSymbols] - `//` or `#` indicates what to use to uncomment
+  void _uncommentSelectedLines(String commentSymbols) {
+    modifySelectedLines((line) {
+      // if line is empty skip it
+      if (RegExps.emptyLine.hasMatch(line)) {
+        return line;
+      }
+
+      // replace first `// ` or `# ` or `//` or `#` with empty string
+      return line.replaceFirst(
+        RegExps.getCommentPlusWhitespaceRegExp(commentSymbols),
+        '',
+      );
```

Comment:
I thinks this is more readable and can be better verified visually than a reg exp:
```suggestion
      for (final sequence in SingleLineComments.byMode[language] ?? []) {
        if (line.trim().startsWith(sequence)) {
          // ...
        }
      }
```